AdyenDropIn/Components/PaymentMethodListComponent.swift

```diff
@@ -56,26 +83,38 @@ internal final class PaymentMethodListComponent: ComponentLoader, PresentableCom
             listItem.imageURL = LogoURLProvider.logoURL(for: component.paymentMethod, environment: apiContext.environment)
             listItem.trailingText = displayInformation.disclosureText
             listItem.subtitle = displayInformation.subtitle
-            listItem.selectionHandler = { [unowned self, unowned component] in
+            listItem.selectionHandler = { [weak self, weak component] in
+                guard let self = self, let component = component else { return }
                 guard !(component is AlreadyPaidPaymentComponent) else { return }
                 self.delegate?.didSelect(component, in: self)
             }
             
+            listItem.deletionHandler = { [weak self, weak component] indexPath, completion in
```

Comment:
this handler is big enough, maybe we can move it to a standalone method?

Reply:
extracted into a function